File: nfcnfc/lib/nfc_managerrrrr/lib_mdr.dart

import 'package:flutter/foundation.dart';
import 'package:nfc_manager/platform_tags.dart';

class LibMDR{
  final _resultOK = 36864;

  int _resultCode(Uint8List result, int len){
    return (result[len - 2].toInt() << 8) + result[len - 1];
  }

  Future<int> isoIsMyCard(IsoDep isoDep, Uint8List cardNo, Uint8List cardInfo)async{
    final apduClSelectApp = Uint8List.fromList([0, -92, 4, 0, 8, 0, 0, 0, 0, 0, 0, 0, 1]);
    final apduClCardInfo = Uint8List.fromList([0, -77, 0, 0, 63]);
    var respData = await isoDep.transceive(data: apduClSelectApp);
    if (_resultCode(respData, respData.length) != _resultOK) {
      return -1;
    }

    respData = await isoDep.transceive(data: apduClCardInfo);
    if (_resultCode(respData, respData.length) != _resultOK) {
      return -1;
    }

    cardNo.setRange(0, 8, respData);
    cardInfo.setRange(0, respData.length - 2, respData);
    return 0;
  }

  Future<int> isoGetFastBalance(IsoDep isoDep, List<int> balance) async {
    Uint8List apduClFastGetBalance = Uint8List.fromList([0, -75, 0, 0, 10]);

    try {
      Uint8List respData = await isoDep.transceive(data: apduClFastGetBalance);
      if (_resultCode(respData, respData.length) != _resultOK) {
        return -1;
      }

      balance[0] = ((respData[3] & 0xff) << 24) |
      ((respData[2] & 0xff) << 16) |
      ((respData[1] & 0xff) << 8) |
      (respData[0] & 0xff);
      return 0;
    } catch (ex) {
      if (kDebugMode) {
        print(ex.toString());
      }
      return -1;
    }
  }
}

File: kotlinnfc/android/app/src/main/kotlin/com/example/kotlinnfc/banklibrary/LibMDR.kt
package com.sardy.nomorkartuuangelektronik.banklibrary

import android.nfc.tech.IsoDep

class LibMDR {

    private val resultOk = -28672

    private fun resultCode(result: ByteArray, len: Int): Int {
        return (result[len - 2].toInt() shl 8) + result[len - 1]
    }

    fun isoIsMyCard(isoDep: IsoDep, cardNo: ByteArray, cardInfo: ByteArray): Int {
        val apduClSelectApp = byteArrayOf(0, -92, 4, 0, 8, 0, 0, 0, 0, 0, 0, 0, 1)
        val apduClCardInfo = byteArrayOf(0, -77, 0, 0, 63)
        try {
            var respData = isoDep.transceive(apduClSelectApp)
            if (resultCode(respData, respData.size) != resultOk) {
                return -1
            }

            respData = isoDep.transceive(apduClCardInfo)
            if (resultCode(respData, respData.size) != resultOk) {
                return -1
            }
            System.arraycopy(respData, 0, cardNo, 0, 8)
            System.arraycopy(respData, 0, cardInfo, 0, respData.size - 2)
            return 0
        } catch (ex: Exception) {
            ex.printStackTrace()
            return -1
        }
    }

    fun isoGetFastBalance(isoDep: IsoDep, balance: IntArray): Int {
        val apduClFastGetBalance = byteArrayOf(0, -75, 0, 0, 10)
        try {
            val respData = isoDep.transceive(apduClFastGetBalance)
            if (resultCode(respData, respData.size) != resultOk) {
                return -1
            }
            balance[0] = ((respData[3].toInt() and 255) shl 24) +
                    ((respData[2].toInt() and 255) shl 16) +
                    ((respData[1].toInt() and 255) shl 8) +
                    (respData[0].toInt() and 255)
            return 0
        } catch (ex: Exception) {
            ex.printStackTrace()
            return -1
        }
    }

}